universal-login-react/src/ui/commons/GasPrice/GasPrice.tsx

```diff
@@ -115,9 +116,18 @@ export const GasPrice = ({isDeployed = true, deployedWallet, sdk, gasLimit, onGa
       </div>
     </>
   );
+
+  const returnRenderComponentOrSpinner = (gasModes?: GasMode[]) => {
+    return (gasModes ? renderComponent(gasModes) : <Spinner className="spinner-small" />);
+  };
+
+  if (sdk?.isRefundPaid()) {
```

Comment:
We should consider making the sdk required property because now it has a big role in the appearance of GasPrice component